sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -287,7 +287,14 @@ internal string QueryCaptionRaw
 
                 if (QueryCaption.HasValue)
                 {
-                    queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
+                    if (QueryCaption.Value != QueryCaptionType.None)
+                    {
+                        queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
+                    }
+                    else
+                    {
+                        queryCaptionStringValue = QueryCaption.Value.ToString();
+                    }
```

Comment:
```suggestion
                    if (QueryCaption.Value == QueryCaptionType.Extractive)
                    {
                        queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
                    }
                    else
                    {
                        queryCaptionStringValue = QueryCaption.Value.ToString();
                    }
```

As per the docs [here](https://learn.microsoft.com/en-us/dotnet/api/azure.search.documents.models.querycaptiontype?view=azure-dotnet-preview#definition):

>When Captions is set to 'extractive', highlighting is enabled by default, and can be configured by appending the pipe character '|' followed by the 'highlight-<true/false>' option, such as 'extractive|highlight-true'. Defaults to 'None'

The condition should be:
```csharp
if (QueryCaption.Value == QueryCaptionType.Extractive)
{
     queryCaptionStringValue = $"{QueryCaption.Value}{QueryCaptionRawSplitter}{QueryCaptionHighlightEnabled.GetValueOrDefault(true)}";
}
else
 {
      queryCaptionStringValue = QueryCaption.Value.ToString();
 }
```